das/dastree/dastree.go

```diff
@@ -43,26 +49,44 @@ func RecordHash(record func(bytes32, []byte), preimage ...[]byte) bytes32 {
 	//  Intermediate hashes like '*' from above may be recorded via the `record` closure
 	//
 
-	keccord := func(value []byte) bytes32 {
+	keccord := func(value []byte) (bytes32, error) {
 		hash := crypto.Keccak256Hash(value)
-		record(hash, value)
-		return hash
+		err := record(hash, value)
+		return hash, err
 	}
 	prepend := func(before byte, slice []byte) []byte {
 		return append([]byte{before}, slice...)
 	}
 
 	unrolled := arbmath.ConcatByteSlices(preimage...)
 	if len(unrolled) == 0 {
-		return arbmath.FlipBit(keccord(prepend(LeafByte, keccord([]byte{}).Bytes())), 0)
+		dataHash, err := keccord([]byte{})
+		if err != nil {
+			return bytes32{}, err
+		}
+
+		leafHash, err := keccord(prepend(LeafByte, dataHash.Bytes()))
+		if err != nil {
+			return bytes32{}, err
+		}
+
+		return arbmath.FlipBit(leafHash, 0), nil
```

Comment:
Same here to remove that `if err != nil`
```go
return arbmath.FlipBit(leafHash, 0), err
```

Too bad this is no longer 1 line but that's just go :/

Reply:
Reverted the changes to the record signature.